File: Codeforces/div811/B.cpp

#include <iostream>
#include <map>
#include <vector>

using namespace::std;

int main()
{
	int t;
	cin >> t;
	while (t--)
	{
		int n;
		cin >> n;
		vector<int> a(n);
		map<int, int> d;
		for (int i=0; i < n; ++i)
			cin >> a.at(i);
		for (int i : a)
		{
			if (d.count(i) == 0)
				d[i] = 0;
			d[i] += 1;
		}
		int remaining = 0;
		for (const auto &i : d)
			if (i.second > 1)
				++remaining;
		int ans = 0;
		for (int i : a)
		{
			if (remaining == 0)
				break ;
			if (d[i] == 2)
			{
				--d[i];
				--remaining;
			}
			if (d[i] > 1)
				--d[i];
			++ans;
		}
		cout << ans << endl;
	}
}

File: Codeforces/div811/B.py

def inp():
    return list(map(int, input().split()))


for _ in range(int(input())):
    n = int(input())
    a = inp()
    d = dict()
    for i in a:
        d[i] = d.get(i, 0) + 1
    remaining = 0
    for i in d.values():
        if i > 1:
            remaining += 1
    ans = 0
    for i in a:
        if remaining == 0:
            break
        if d[i] == 2:
            d[i] -= 1
            remaining -= 1
        if d[i] > 1:
            d[i] -= 1
        ans += 1
    print(ans)

